doc/Data Plane Code Generation/Subclients.md

```diff
@@ -0,0 +1,247 @@
+# Customizing the generated code
+
+## Subclients
+
+There are two categories of clients: service clients and their subclients. Service clients can be instantiated and have the Client suffix. Subclients can only be created by calling factory methods on other clients (commonly on service clients) and do not have the client suffix. The service client is the entry point to the API for an Azure service. Please refer to [guideline](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients) for more detail information.
+
+There are two tools to organize subclients and service client.
+
+- **Single top-level client**: define one top-level service client with the name <service_name>Client, all other clients will be its subclient.
+
+- **client hierarchy**: organize the client service and subclients via defining the parent-subclient hierarchy dependency.
+
+**Note**: You can combine those two approaches to define an client hierarchy with one top-level client and its subclient may have sub clients also. 
+
+### Single top-level client
+
+With multiple service clients it may be better to group them all under one top-level client.
+
+<details>
+
+**Generated code before:**
+
+``` C#
+//Generated\RootClient.cs
+namespace Azure.Service.SubClients
+{
+    public partial class RootClient
+    {
+        public RootClient(string cachedParameter, AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
+    }
+}
+
+//Generated\ParameterClient.cs
+namespace Azure.Service.SubClients
+{
+    public partial class ParameterClient
+    {
+        public ParameterClient(AzureKeyCredential credential, Uri endpoint = null, SubClientsClientOptions options = null){}
+    }
+}
+```
+
+**Add customize configuration (set single-top-level-client):**
+
+``` md
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+``` yaml
+namespace: Azure.Service.SubClients
+input-file: $(this-folder)/SubClients.json
+data-plane: true
+security: AzureKey
+security-header-name: Fake-Subscription-Key
+single-top-level-client: true
```

Comment:
It would be good to highlight that this is the setting that adds the client and not the others.  Is there a way to specify the name of the top-level client?

Reply:
No. It is default to use the ServiceClient as the to-level client, the service client name is the title of the client

Reply:
And I don't find a way to highlight a line in an ymal code snippet. @annelo-msft Do you know any way to do so? Thanks

Reply:
See https://github.com/Azure/azure-sdk-for-net/pull/27559#discussion_r838820230

Reply:
I don't know for yaml specifically - is this something you could investigate?